extensions/azure/storage/AzureDataLakeStorageClient.cpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file AzureDataLakeStorageClient.cpp
+ * AzureDataLakeStorageClient class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "AzureDataLakeStorageClient.h"
+
+namespace org::apache::nifi::minifi::azure::storage {
+
+void AzureDataLakeStorageClient::resetClientIfNeeded(const std::string& connection_string, const std::string& file_system_name) {
```

Comment:
it seems to me that a single `AzureDataLakeStorage` instance is contained in the processors, two parallel threads of which might call `resetClientIfNeeded` simultaneously without locks, with possibly different `file_system_name`

Reply:
Every processor has its own `AzureDataLakeStorage` instance containing separate clients for each processor, there should be no problems with concurrency.

Reply:
Do you think we should implement a mechanism to process the requests in parallel until a client reset is needed (then wait for the ongoing parallel requests to finish and reset the client continuing to process the queued requests) or sequential processing with a lock is sufficient?

Reply:
in light of the recent discussions it is okay if we restrict the processor to a single thread by a mutex